f8a_worker/workers/graph_importer_utils.py

```diff
@@ -0,0 +1,204 @@
+                    ver_obj = _other_key_info(contents.get('ver_list_keys'), "version")
+                    if 'analyses' in obj:
+                        obj.get('analyses', {}).update(ver_obj['analyses'])
+                    else:
+                        obj.update(ver_obj)
+
+                # Check Package related information and add it to package object
+                if len(contents.get('pkg_list_keys')) > 0:
+                    pkg_obj = _other_key_info(contents.get('pkg_list_keys'), "package")
+                    if 'analyses' in obj:
+                        obj.get('analyses', {}).update(pkg_obj['analyses'])
+                    else:
+                        obj.update(pkg_obj)
+
+                # Create Gremlin Query
+                str_gremlin = GraphPopulator.create_query_string(obj)
+
+                if str_gremlin:
+                    # Fire Gremlin HTTP query now
+                    logger.info("Ingestion initialized for EPV - " +
+                                obj.get('ecosystem') + ":" + obj.get('package') + ":" + obj.get('version'))
+                    epv.append(obj.get('ecosystem') + ":" + obj.get('package') + ":" + obj.get('version'))
+                    payload = {'gremlin': str_gremlin}
+                    response = requests.post(GREMLIN_SERVER_URL_REST, data=json.dumps(payload), timeout=30)
+                    resp = response.json()
+
+                    if 'status' in resp and resp['status']['code'] == 200:
+                        count_imported_EPVs += 1
+                        last_imported_EPV = obj.get('ecosystem') + ":" + obj.get('package') + ":" + obj.get('version')
+                    elif 'Exception-Class' in resp:
+                        report['message'] = "The import failed " + resp['message']
+                        report['status'] = 'Failure'
+
+            except Exception as e:
+                msg = _get_exception_msg("The import failed", e)
+                report['status'] = 'Failure'
+                report['message'] = msg
+                report['epv'] = epv_key
+
+    report['epv'] = epv_list
+    report['count_imported_EPVs'] = count_imported_EPVs
+    if count_imported_EPVs == 0 and report['status'] == 'Success':
+        report['message'] = 'Nothing to be synced to Graph!'
+    report['last_imported_EPV'] = last_imported_EPV
+
+    return report
+
+
+def import_epv_from_s3_http(list_epv, select_doc=None):
+
+    try:
+        global s3, s3_pkg
+        s3 = StoragePool.get_connected_storage('S3Data')
+        s3_pkg = StoragePool.get_connected_storage('S3PackageData')
+        # Collect relevant files from data-source and group them by package-version.
+        list_keys = []
+        for epv in list_epv:
+            dict_keys = {}
+            ver_list_keys = []
+            pkg_list_keys = []
+            pkg_key_prefix = ver_key_prefix = ""
+
+            if 'name' not in epv or 'ecosystem' not in epv:
+                continue
```

Comment:
yes it is only one EPV from worker. But, I would still prefer to let it be, just in future, we may want to bypass all except graph syncs.

Reply:
I would rather keep code minimal, easy to maintain, easy to extend and easy to read without adding unnecessary complexity that could easily lead to errors. We can still get back to the previous version based on git logs (see recent "reintroduction" of go support). WDYT?

Reply:
I personally feel we should leave this functionality intact at least for the time being, this gives us flexibility in the future to just invoke graph syncs by-passing all other tasks.. Let's ingest one ecosystem with graph sync and check whether this would make any sense.